tests/test_types.py

```diff
@@ -31,4 +41,63 @@ def test_count(count, truth):
 def test_dict():
     now = dt.datetime.utcnow()
     model = QueryParams(count=1, created_before=now)
-    assert model.dict() == dict(count=1, created_before=now.isoformat())
+    assert model.dict() == dict(count=1, created_before=now.isoformat() + 'Z')
+
+
+def test_json_encoder():
+    class EnumTest(Enum):
+        s, p, e, i, d = range(5)
+
+    @dataclass
+    class TestClass:
+        uno: str
+
+        def to_dict(self):
+            return dict(uno=self.uno, dos='dos')
+
+    now = dt.datetime.utcnow()
+    test_class = TestClass(uno='uno')
+
+    to_encode = dict(enum=EnumTest.s, now=now, test_class=test_class,)
+
+    encoded = json.dumps(to_encode, cls=CJSONEncoder)
+    decoded = json.loads(encoded)
+
+    assert decoded['enum'] == 0
+    assert decoded['now'] == now.isoformat() + 'Z'
+    assert decoded['test_class']['uno'] == 'uno'
+    assert decoded['test_class']['dos'] == 'dos'
+
+
+def test_invalid_class():
+    class ClassWithoutToDict:
+        ...
+
+    invalid_class = ClassWithoutToDict()
+    with pytest.raises(TypeError):
```

Comment:
What are you actually testing? Why is it raising `TypeError`?

Reply:
It's a test for the case when the item is not a `dt.date` nor `Enum` and it doesn't contain a method `to_dict`, so it uses the default `json.JSONEncoder` method which raises that exception. I'll add a comment to make it clearer